### PR TITLE
Unify EFB encoding shader generation

### PR DESCRIPTION
**Pull request by magumagu**
Per title, delete a bunch of D3D-specific code, and use the code in VideoCommon.  Still not doing coordinate math quite right, and I broke OpenGL in the process of fixing D3D, but the idea should be clear from this patch.

### CODE REVIEW

**Comment by magumagu** [comment]
@dolphin-emu-bot rebuild

**Comment by magumagu** [comment]
No longer WIP: I think I've managed to track down all the remaining issues.

File: Source/Core/VideoCommon/TextureConversionShader.cpp [hunk 1]
```diff
@@ -70,36 +70,38 @@ static void WriteSwizzler(char*& p, u32 format, API_TYPE ApiType)
...
-	WRITE(p, "  int x_block_position = (x_virtual_position >> %d) & %d;\n", IntLog2(blkH), ~(blkW - 1));
+	WRITE(p, "  int x_block_position = (uv1.x >> %d) << %d;\n", IntLog2(blkH * blkW / samples), IntLog2(blkW));
+	WRITE(p, "  int y_block_position = uv1.y << %d;\n", IntLog2(blkH));
 	if (samples == 1)
 	{
-		// 32 bit textures (RGBA8 and Z24) are stored in 2 cache line increments
-		WRITE(p, "  bool first = 0 == (x_virtual_position & %d);\n", 8 * samples); // first cache line, used in the encoders
-		WRITE(p, "  x_virtual_position = x_virtual_position << 1;\n");
+		// With samples == 1, we write out pairs of blocks; one A8R8, one G8B8.
+		WRITE(p, "  bool first = !(uv1.x & %d);\n", blkH * blkW / 2);
```
Comment (Sonicadvance1) [inline Source/Core/VideoCommon/TextureConversionShader.cpp]: Boolean negation is broken on Adreno.

**Comment by magumagu** [comment]
Pushed a fix for the Boolean negation and another issue I discovered.

File: Source/Core/VideoBackends/OGL/TextureConverter.cpp [hunk 2]
```diff
@@ -35,7 +35,7 @@ static GLuint s_texConvFrameBuffer[2] = {0,0};
 static GLuint s_srcTexture = 0; // for decoding from RAM
 static GLuint s_dstTexture = 0; // for encoding to RAM
 
-const int renderBufferWidth = 1024;
+const int renderBufferWidth = EFB_WIDTH * 4;
 const int renderBufferHeight = 1024;
```
Comment (degasus) [inline Source/Core/VideoBackends/OGL/TextureConverter.cpp]: Isn't he new hight also EFB_HEIGHT / 4 ?

**Comment by degasus** [comment]
LGTM, but I think this PR needs more testing, both for performance and for correctness.

File: Source/Core/VideoBackends/OGL/TextureConverter.cpp (hunk 2 above)
Comment (magumagu) [inline Source/Core/VideoBackends/OGL/TextureConverter.cpp]: Yes... except that we also use this buffer for XFB encoding, which requires substantially more height.  If it matters, I can change it to use two separate buffers.

**Comment by degasus** [comment]
looks very fine on ogl here

**Pull request merged**